js/src/volume.js

```diff
@@ -279,6 +282,20 @@ var ScatterView = widgets.WidgetView.extend( {
         _.mapObject(this.model.changedAttributes(), function(val, key){
             console.log("changed " +key)
             this.previous_values[key] = this.model.previous(key)
+            if (key == "index"){
+              pindex = this.previous_values["index"]
+              this.previous_values["x"] = this.model.previous("x")[pindex]
+              this.previous_values["y"] = this.model.previous("y")[pindex]
+              this.previous_values["z"] = this.model.previous("z")[pindex]
```

Comment:
Not every x, y, z, vx, vy, vz needs to be a sequence, can also be that just x is a sequence of arrays, and the rest is not. This will happen when you do updates from the notebook like
```python
scatter.x = x2d
scatter.y = y2d
```
They will be updated in sequence.
Also, I think it should be:
```
if(_.isArray(this.model.get('x')) { // changing index effectively changes the 'x' that the shader sees
  this.previous_values["x"] = this.model.get('x')[this.model.previous('sequence_index')]
  this.attributes_changed['x'] = ['x']
} // else by changing the sequence_index you don't really change x
```